File: Daily Test/C++/Chess Piece - Validate Move 17.3.2026.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    char a;
    string b, c;
    cin >> a >> b >> c;

    int r = b[1] - '0';
    int col = b[0] - 'a';

    int tr = c[1] - '0';
    int tc = c[0] - 'a';

    vector<pair<int,int>> moves;

    if(a == 'R'){
        for(int i=0;i<8;i++){
            moves.push_back({r,i});
            moves.push_back({i,col});
        }
    }
    else if(a == 'B'){
        for(int i=1;i<8;i++){
            if(r+i<8 && col+i<8) moves.push_back({r+i,col+i});
            if(r+i<8 && col-i>=0) moves.push_back({r+i,col-i});
            if(r-i>=0 && col+i<8) moves.push_back({r-i,col+i});
            if(r-i>=0 && col-i>=0) moves.push_back({r-i,col-i});
        }
    }
    else{
        int dr[] = {2,2,-2,-2,1,1,-1,-1};
        int dc[] = {1,-1,1,-1,2,-2,2,-2};

        for(int i=0;i<8;i++){
            int nr=r+dr[i], nc=col+dc[i];
            if(nr>=0 && nr<8 && nc>=0 && nc<8)
                moves.push_back({nr,nc});
        }
    }

    bool found=false;
    for(auto m:moves){
        if(m.first==tr && m.second==tc){
            found=true;
            break;
        }
    }

    cout<<(found?"Yes":"No");
}

File: Daily Test/Java/Chess Piece - Validate Move 17.3.2026.java
import java.util.*;

public class Main {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        char a = sc.next().charAt(0);
        String b = sc.next();
        String c = sc.next();

        int r = b.charAt(1) - '0';
        int col = b.charAt(0) - 'a';

        int tr = c.charAt(1) - '0';
        int tc = c.charAt(0) - 'a';

        List<int[]> moves = new ArrayList<>();

        if(a == 'R'){
            for(int i=0;i<8;i++){
                moves.add(new int[]{r,i});
                moves.add(new int[]{i,col});
            }
        }
        else if(a == 'B'){
            for(int i=1;i<8;i++){
                if(r+i<8 && col+i<8) moves.add(new int[]{r+i,col+i});
                if(r+i<8 && col-i>=0) moves.add(new int[]{r+i,col-i});
                if(r-i>=0 && col+i<8) moves.add(new int[]{r-i,col+i});
                if(r-i>=0 && col-i>=0) moves.add(new int[]{r-i,col-i});
            }
        }
        else{
            int[] dr = {2,2,-2,-2,1,1,-1,-1};
            int[] dc = {1,-1,1,-1,2,-2,2,-2};

            for(int i=0;i<8;i++){
                int nr=r+dr[i], nc=col+dc[i];
                if(nr>=0 && nr<8 && nc>=0 && nc<8)
                    moves.add(new int[]{nr,nc});
            }
        }

        boolean found=false;
        for(int[] m:moves){
            if(m[0]==tr && m[1]==tc){
                found=true;
                break;
            }
        }

        System.out.println(found?"Yes":"No");
    }
}